lib/web_ui/lib/src/ui/geometry.dart

```diff
@@ -859,7 +859,7 @@ class Rect {
   ///
   /// Values for `t` are usually obtained from an [Animation<double>], such as
   /// an [AnimationController].
-  static Rect/*?*/ lerp(Rect/*?*/ a, Rect/*?*/ b, double/*!*/ t) {
+  static Rect? lerp(Rect? a, Rect? b, double t) {
     assert(t != null);
     if (b == null) {
```

Comment:
Ok, I just have to say - how cool is it that this code just worked with no changes?

Reply:
I did change this function in a previous PR to assist with type inference: https://github.com/flutter/engine/commit/f581f428e981d810425f33ef0035fea8cf59a0df#diff-3afaf9265618b9fee97b10a7de58cc76

But it's still pretty cool that despite both `a` and `b` being nullable, there are no `!` in the body of the function.

Reply:
Ah, I almost asked why it wasn't written the way it was pre-refactor... :)

If avoiding the extra instructions is important, you could write it as:

```
if (a == null) return (b == null) ? null : b * t;
if (b == null) return (a == null) ? null : a * (1.0 - t)
return ....
```

I also noticed that there were places where this was called with what looked like definitely non-nullable arguments, which made me wonder whether it was worth having a specialized version.  But I have no idea which pieces of this are perf sensitive, so...

Reply:
It would be nice to have an annotation instructing the compiler to specialize automatically, similar to [Scala's @specialized](https://www.scala-lang.org/api/current/scala/specialized.html).

Reply:
Done: https://github.com/dart-lang/sdk/issues/42361